### PR TITLE
Do not throw an exception if a module is invalid

### LINKED ISSUE #28305: A not installed module still impact the Backoffice
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

1 - Install MBO module
[ps_mbo (15).zip](https://github.com/PrestaShop/PrestaShop/files/8539338/ps_mbo.15.zip)

2 - Go to Modules > Modules catalog and search for "Gamiphy" in the list

OR 

Download Gamiphy from https://addons.prestashop.com/fr/fidelisation-parrainage/52050--gamiphy-fidelisation-avec-jeux-et-parrainages-.html

3 - Try to install the module => it fails

4 - Refresh the page, you have an error related to Gamiphy module
image-2022-04-20-14-43-51-440

### Expected behavior

If the installation fails, the module's files have to be deleted or at least not loaded

### Steps to reproduce

See issue description

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR aims to prevent exception to be thrown if a module is invalid.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28305
| Related PRs       | 
| How to test?      | Please see #28305
| Possible impacts? |

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: src/Core/Module/ModuleRepository.php [hunk 1]
```diff
@@ -150,6 +151,10 @@ public function getModule(string $moduleName): ModuleInterface
 
         $isValid = $filemtime > 0 && $this->moduleDataProvider->isModuleMainClassValid($moduleName);
         $attributes = $this->getModuleAttributes($moduleName, $isValid);
+        if (empty($attributes)) {
+            $isValid = false;
+        }
+        $attributes = array_merge(['name' => $moduleName], $attributes);
```
Comment (PululuK) [inline src/Core/Module/ModuleRepository.php:157]:
```suggestion
        $attributes['name'] = $moduleName;
```

Comment (PululuK) [reply src/Core/Module/ModuleRepository.php:157]: Hi @atomiix 
Why array merge ?

Comment (atomiix) [reply src/Core/Module/ModuleRepository.php:157]: @PululuK, why not? 😄

Comment (PululuK) [reply src/Core/Module/ModuleRepository.php:157]: 😁 yep why not ... But my question was really sincere, I really wanted to understand. Maybe you didn't want to overwrite the name key if it already existed ?

Comment (atomiix) [reply src/Core/Module/ModuleRepository.php:157]: Nothing more than personal preference 😄 I think the purpose is clearer here when we use `array_merge` than just an new assignment. Also, it puts `name` as the first key of the array (easier to debug).

Comment (PululuK) [reply src/Core/Module/ModuleRepository.php:157]: Okay but I think a new assignment is faster than array_merge (involves more processing), of course it's micro optimizations, but when it goes in a loop we can save a few seconds.

Comment (atomiix) [reply src/Core/Module/ModuleRepository.php:157]: I saw in [this thread](https://stackoverflow.com/questions/57192782/array-merge-vs-direct-array-injection-performance), that `array_merge` was faster 🤷‍♂️ 😅

Comment (PululuK) [reply src/Core/Module/ModuleRepository.php:157]: It's not in the same context, in your case :

- array has only one element(equivalent to add a simple variable and assigning a value to it)

Comment (PululuK) [reply src/Core/Module/ModuleRepository.php:157]: According the thread https://stackoverflow.com/questions/57192782/array-merge-vs-direct-array-injection-performance
image

Remember that each "if" in this sentence represents a calculation

Comment (atomiix) [reply src/Core/Module/ModuleRepository.php:157]: You're right!
But as you said, this is micro optimization, I think we can keep it?

**Review by matks — DISMISSED** [review]
OK for me. I let @atomiix and @PululuK run extensive benchmarks to decide what to do with the array operation 🤣

**Review by marwachelly — COMMENTED** [review]
Hello @atomiix,

I tested the PR on develop branch with PHP version: 8.0.17 and MySQL version: 5.7.36. But the issue is still not fixed. There still a regression on the page module catalog after installing the MBO module.

https://user-images.githubusercontent.com/89604350/172655256-39c3aa30-dcbe-444c-922c-4716979fd48f.mp4

Could you please check?

Thanks,

**Comment by atomiix** [comment]
Hi @marwachelly, it looks like the bug happens in the module catalog page but not in the module manager page anymore.
I think this is something to fix in the ps_mbo module.

**Comment by marwachelly** [comment]
> Hi @marwachelly, it looks like the bug happens in the module catalog page but not in the module manager page anymore. I think this is something to fix in the ps_mbo module.

As discussed in the meet. PR needs more investigation from @atomiix.

Many thanks @atomiix :blush:

**Comment by marwachelly** [comment]
Hello @sowbiba,

The problem is fixed on the page module manager but the same exception is observed when clicking on the page module catalogue. What do you think? 

https://user-images.githubusercontent.com/89604350/172655256-39c3aa30-dcbe-444c-922c-4716979fd48f.mp4

Thanks,

**Comment by atomiix** [comment]
@sowbiba friendly reminder ;)

**Review by jolelievre — DISMISSED** [review]

**Comment by jolelievre** [comment]
Hi @atomiix 

I'm not sure to understand the status of this PR/issue should it be fixed in MBO by @sowbiba or in the core by you?

Also, I believe the PR was created before feature freeze, should it be rebased on 8.0.x and we should change the target of the PR maybe?

**Comment by sowbiba** [comment]
We catch the exception to avoid it to be displayed but the module files are still "loaded"

It's fine for now, at least the customer is not blocked ;)

**Review by sowbiba — DISMISSED** [review]
We catch the exception to avoid it to be displayed but the module files are still "loaded"

It's fine for now, at least the customer is not blocked ;)

**Review by sowbiba — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by marwachelly — COMMENTED** [review]
Hello @atomiix,

I tested the PR on the branch 8.0.x but I still see the same[ issue](https://github.com/PrestaShop/PrestaShop/pull/28416#issuecomment-1151322866) when clicking on the page module catalog. 

https://user-images.githubusercontent.com/89604350/180035879-62863c59-55cc-432c-97b3-1d296eaac888.mp4

@sowbiba what do you think?

Thanks,

**Comment by atomiix** [comment]
@marwachelly yes, that's what we discussed with @sowbiba, the bug must be fixed in the module catalog but this PR is about the module manager ;)

**Review by marwachelly — APPROVED** [review]
Hello @atomiix,

The issue is well fixed in the module manager and the automatic tests are green. So PR is QA approved. 

Thanks,

**Comment by kpodemski** [comment]
Thanks @atomiix @marwachelly

**Pull request merged**